src/lib.rs

```diff
@@ -0,0 +1,100 @@
+//! Pull request management for bare repos
+use std::io;
+use std::ffi::OsString;
+use std::path::Path;
+use std::process::Command;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+#[derive(Debug)]
+pub struct Git {
+    program: OsString,
+}
+
+impl Git {
+    pub fn new() -> Git {
+        Git{ program: OsString::from("git") }
+    }
+    pub fn with_path(path: &Path) -> Git {
+        Git{ program: path.as_os_str().to_os_string() }
+    }
+    pub fn version(&self) -> io::Result<String> {
+        let output = Command::new(&self.program).arg("--version").output()?;
+        if !output.status.success() {
+            return Err(io::Error::new(io::ErrorKind::Other, "git --version"));
+        }
+        Ok(String::from_utf8_lossy(&output.stdout).to_string())
+    }
+    pub fn fetch_prune(&self) -> io::Result<()> {
+        let status = Command::new(&self.program).args(&["fetch","--prune"]).status()?;
+        if !status.success() {
+            return Err(io::Error::new(io::ErrorKind::Other, "git fetch --prune"));
+        }
+        Ok(())
+    }
+    pub fn all_branches(&self) -> io::Result<String> {
+        let output = Command::new(&self.program).args(&["branch","-a"]).output()?;
+        if !output.status.success() {
+            return Err(io::Error::new(io::ErrorKind::Other, "git branch -a"));
+        }
+        Ok(String::from_utf8_lossy(&output.stdout).to_string())
+    }
+}
+
+pub fn extract_pr_names(branches: &str) -> Vec<String> {
+    lazy_static! {
+        static ref BEGINS_WITH_REMOTE_REF: Regex = Regex::new(r"^ *\** remotes/origin/").unwrap();
```

Comment:
Using origin can cause issues with others not using that wordage. Replace with "not slash" match as it can be anything.

Reply:
Yeah, that works for me.

Reply:
Your original proposal was best: https://stackoverflow.com/questions/41461152/which-characters-are-illegal-within-a-git-remote-name/41462742

git doesn't explicitly say what characters can't be in a remote name, so the only thing we can really do is consume every character until the first '/', which is what c0a8dae6bd6ebaa6e261e9f97c1d18f97ecbcd66 does.